### PR TITLE
Fix problems in JibriDetector on XMPP reconnect

### PR DESCRIPTION
**Pull request by paweldomas**
1. fixes some threads dying while trying to send packets on broken XMPP connection(may lead to weird state left)
2. fixes invalid Jibri addresses kept in the memory after XMPP reconnection

### CODE REVIEW

File: src/main/java/org/jitsi/jicofo/recording/jibri/JibriRecorder.java [hunk 1]
```diff
@@ -197,10 +197,19 @@ public void init()
...
+            {
+                logger.error(
+                    "An error response to the stop request: " + error.toXML());
+            }
+        }
+        catch (OperationFailedException e)
         {
-            logger.error("Error when sending stop request: " + error.toXML());
+            logger.error("Failed to send stop IQ - XMPP disconnected", e);
+            recordingStopped(null, false /* do not send any status updates */);
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jicofo/recording/jibri/JibriRecorder.java]: Is it intentional that we call recordingStopped on exeption, but not when "error != null" above?

File: src/main/java/org/jitsi/jicofo/recording/jibri/JibriDetector.java [hunk 2]
```diff
@@ -222,6 +222,14 @@ private void stop()
             chatRoom = null;
 
             logger.info("Left JIBRI room: " + jibriBrewery);
+
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jicofo/recording/jibri/JibriDetector.java]: Is there any particular reason this only executes if "chatRoom != null"?

Comment (paweldomas) [inline src/main/java/org/jitsi/jicofo/recording/jibri/JibriDetector.java]: No, will fix that, thanks !

File: src/main/java/org/jitsi/jicofo/recording/jibri/JibriRecorder.java (hunk 1 above)
Comment (paweldomas) [inline src/main/java/org/jitsi/jicofo/recording/jibri/JibriRecorder.java]: It got kind of complex now. Will take a second look at this.

**Comment by bgrozev** [comment]
Jenkins: ok to test

**Pull request merged**